notebooks/214-grammar-correction/214-grammar-correction.ipynb

```diff
@@ -0,0 +1,500 @@
+    "    separator (str, optional): The separator to use when joining the corrected text into a single string. Default is a space character.\n",
+    "\n",
+    "    Returns:\n",
+    "    str: The corrected text.\n",
+    "    \"\"\"\n",
+    "    # Split the text into sentence batches\n",
+    "    sentence_batches = split_text(text)\n",
+    "\n",
+    "    # Initialize a list to store the corrected text\n",
+    "    corrected_text = []\n",
+    "\n",
+    "    # Iterate through the sentence batches\n",
+    "    for batch in tqdm(\n",
+    "        sentence_batches, total=len(sentence_batches), desc=\"correcting text..\"\n",
+    "    ):\n",
+    "        # Join the sentences in the batch into a single string\n",
+    "        raw_text = \" \".join(batch)\n",
+    "\n",
+    "        # Check the grammar quality of the text using the text-classification pipeline\n",
+    "        results = checker(raw_text)\n",
+    "\n",
+    "        # Only correct the text if the results of the text-classification are not LABEL_1 or are LABEL_1 with a score below 0.9\n",
+    "        if results[0][\"label\"] != \"LABEL_1\" or (\n",
+    "            results[0][\"label\"] == \"LABEL_1\" and results[0][\"score\"] < 0.9\n",
+    "        ):\n",
+    "            # Correct the text using the text-generation pipeline\n",
+    "            corrected_batch = corrector(raw_text)\n",
+    "            corrected_text.append(corrected_batch[0][\"generated_text\"])\n",
+    "        else:\n",
+    "            corrected_text.append(raw_text)\n",
+    "\n",
+    "    # Join the corrected text into a single string\n",
+    "    corrected_text = separator.join(corrected_text)\n",
+    "\n",
+    "    return corrected_text"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "26d3d759-3cb2-418d-82f8-3be2e445916a",
+   "metadata": {},
+   "source": [
+    "Let's look at it in action. Put the text which should be corrected into the text box and execute the next cells."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "aee397f5-12cb-460b-8824-327f19af8e5f",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "05ab1e637f0546b1900ce30bce384bd4",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Textarea(value='Most of the course is about semantic or  content of language but there are also interesting to…"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import ipywidgets as widgets\n",
+    "\n",
+    "style = {'description_width': 'initial'}\n",
+    "\n",
+    "text_widget = widgets.Textarea(value=\"Most of the course is about semantic or  content of language but there are also interesting topics to be learned from the servicefeatures except statistics in characters in documents.\"\n",
```

Comment:
Is there any way to make the default textbox size bigger? Does the widget allow it? 
<img width="277" alt="image" src="https://user-images.githubusercontent.com/15709723/220488806-8fa5a8d7-112a-4ccc-8c87-8e013273d6f2.png">

Reply:
addad layout auto